"""
author: EDF Energy R&D UKC
"""
import sys
import os
import subprocess
import inspect
from PyQt5 import QtCore, QtGui, QtWidgets
from PyQt5.QtWidgets import QApplication, QFileDialog, QDialog, QPlainTextEdit
from PyQt5.QtWidgets import QMainWindow

class UiModuleMeshing(object):
    """
    Meshing gui module
    """

    def setupUi(self, Form):
        Form.setObjectName("Form")
        Form.resize(829, 665)
        self.weld_label = QtWidgets.QLabel(Form)
        self.weld_label.setGeometry(QtCore.QRect(10, 10, 141, 18))
        self.weld_label.setObjectName("weld_label")
        self.weld = QtWidgets.QComboBox(Form)
        self.weld.setGeometry(QtCore.QRect(160, 10, 121, 26))
        self.weld.setObjectName("weld")
        self.weld.addItem("")
        self.weld.addItem("")
        self.weld.addItem("")
        self.stackedWidget = QtWidgets.QStackedWidget(Form)
        self.stackedWidget.setGeometry(QtCore.QRect(20, 50, 781, 561))
        self.stackedWidget.setObjectName("stackedWidget")
        self.home = QtWidgets.QWidget()
        self.home.setObjectName("home")
        self.label = QtWidgets.QLabel(self.home)
        self.label.setGeometry(QtCore.QRect(20, 50, 321, 141))
        self.label.setObjectName("label")
        self.stackedWidget.addWidget(self.home)
        self.TG9 = QtWidgets.QWidget()
        self.TG9.setObjectName("TG9")
        self.b_label_2 = QtWidgets.QLabel(self.TG9)
        self.b_label_2.setGeometry(QtCore.QRect(0, 90, 141, 18))
        self.b_label_2.setObjectName("b_label_2")
        self.b_2 = QtWidgets.QTextEdit(self.TG9)
        self.b_2.setGeometry(QtCore.QRect(180, 80, 51, 31))
        self.b_2.setObjectName("b_2")
        self.a_2 = QtWidgets.QTextEdit(self.TG9)
        self.a_2.setGeometry(QtCore.QRect(180, 40, 51, 31))
        self.a_2.setObjectName("a_2")
        self.c_2 = QtWidgets.QTextEdit(self.TG9)
        self.c_2.setGeometry(QtCore.QRect(180, 120, 51, 31))
        self.c_2.setObjectName("c_2")
        self.a_label_2 = QtWidgets.QLabel(self.TG9)
        self.a_label_2.setGeometry(QtCore.QRect(0, 50, 171, 18))
        self.a_label_2.setObjectName("a_label_2")
        self.c_label_2 = QtWidgets.QLabel(self.TG9)
        self.c_label_2.setGeometry(QtCore.QRect(0, 130, 151, 18))
        self.c_label_2.setObjectName("c_label_2")
        self.label_5 = QtWidgets.QLabel(self.TG9)
        self.label_5.setGeometry(QtCore.QRect(0, 0, 111, 20))
        font = QtGui.QFont()
        font.setBold(True)
        font.setWeight(75)
        self.label_5.setFont(font)
        self.label_5.setObjectName("label_5")
        self.d_label_2 = QtWidgets.QLabel(self.TG9)
        self.d_label_2.setGeometry(QtCore.QRect(0, 170, 171, 18))
        self.d_label_2.setObjectName("d_label_2")
        self.d_2 = QtWidgets.QTextEdit(self.TG9)
        self.d_2.setGeometry(QtCore.QRect(180, 160, 51, 31))
        self.d_2.setObjectName("d_2")
        self.f_label_2 = QtWidgets.QLabel(self.TG9)
        self.f_label_2.setGeometry(QtCore.QRect(0, 250, 151, 18))
        self.f_label_2.setObjectName("f_label_2")
        self.e_label_2 = QtWidgets.QLabel(self.TG9)
        self.e_label_2.setGeometry(QtCore.QRect(0, 210, 151, 18))
        self.e_label_2.setObjectName("e_label_2")
        self.e_2 = QtWidgets.QTextEdit(self.TG9)
        self.e_2.setGeometry(QtCore.QRect(180, 200, 51, 31))
        self.e_2.setObjectName("e_2")
        self.f_2 = QtWidgets.QTextEdit(self.TG9)
        self.f_2.setGeometry(QtCore.QRect(180, 240, 51, 31))
        self.f_2.setObjectName("f_2")
        self.label_6 = QtWidgets.QLabel(self.TG9)
        self.label_6.setGeometry(QtCore.QRect(0, 310, 111, 20))
        font = QtGui.QFont()
        font.setBold(True)
        font.setWeight(75)
        self.label_6.setFont(font)
        self.label_6.setObjectName("label_6")
        self.label_7 = QtWidgets.QLabel(self.TG9)
        self.label_7.setGeometry(QtCore.QRect(310, 30, 371, 241))
        self.label_7.setText("")
        self.label_7.setPixmap(QtGui.QPixmap("TG9.png"))
        self.label_7.setScaledContents(True)
        self.label_7.setObjectName("label_7")
        self.stackedWidget.addWidget(self.TG9)
        self.Edge_Weld = QtWidgets.QWidget()
        self.Edge_Weld.setObjectName("Edge_Weld")
        self.a = QtWidgets.QTextEdit(self.Edge_Weld)
        self.a.setGeometry(QtCore.QRect(180, 40, 51, 31))
        self.a.setObjectName("a")
        self.edge_weld_image = QtWidgets.QLabel(self.Edge_Weld)
        self.edge_weld_image.setGeometry(QtCore.QRect(240, 0, 381, 201))
        self.edge_weld_image.setText("")
        self.edge_weld_image.setPixmap(QtGui.QPixmap("EdgeWelded.png"))
        self.edge_weld_image.setScaledContents(True)
        self.edge_weld_image.setObjectName("edge_weld_image")
        self.c_label = QtWidgets.QLabel(self.Edge_Weld)
        self.c_label.setGeometry(QtCore.QRect(0, 130, 151, 18))
        self.c_label.setObjectName("c_label")
        self.label_1 = QtWidgets.QLabel(self.Edge_Weld)
        self.label_1.setGeometry(QtCore.QRect(0, 0, 111, 20))
        font = QtGui.QFont()
        font.setBold(True)
        font.setWeight(75)
        self.label_1.setFont(font)
        self.label_1.setObjectName("label_1")
        self.c = QtWidgets.QTextEdit(self.Edge_Weld)
        self.c.setGeometry(QtCore.QRect(180, 120, 51, 31))
        self.c.setObjectName("c")
        self.a_label = QtWidgets.QLabel(self.Edge_Weld)
        self.a_label.setGeometry(QtCore.QRect(0, 50, 171, 18))
        self.a_label.setObjectName("a_label")
        self.b = QtWidgets.QTextEdit(self.Edge_Weld)
        self.b.setGeometry(QtCore.QRect(180, 80, 51, 31))
        self.b.setObjectName("b")
        self.b_label = QtWidgets.QLabel(self.Edge_Weld)
        self.b_label.setGeometry(QtCore.QRect(0, 90, 141, 18))
        self.b_label.setObjectName("b_label")
        self.d = QtWidgets.QTextEdit(self.Edge_Weld)
        self.d.setGeometry(QtCore.QRect(180, 160, 51, 31))
        self.d.setObjectName("d")
        self.d_label = QtWidgets.QLabel(self.Edge_Weld)
        self.d_label.setGeometry(QtCore.QRect(0, 170, 151, 18))
        self.d_label.setObjectName("d_label")
        self.edge_weld_mesh = QtWidgets.QLabel(self.Edge_Weld)
        self.edge_weld_mesh.setGeometry(QtCore.QRect(240, 270, 531, 211))
        self.edge_weld_mesh.setText("")
        self.edge_weld_mesh.setPixmap(QtGui.QPixmap("EdgeWeldedMesh.png"))
        self.edge_weld_mesh.setScaledContents(True)
        self.edge_weld_mesh.setObjectName("edge_weld_mesh")
        self.label_2 = QtWidgets.QLabel(self.Edge_Weld)
        self.label_2.setGeometry(QtCore.QRect(0, 230, 111, 20))
        font = QtGui.QFont()
        font.setBold(True)
        font.setWeight(75)
        self.label_2.setFont(font)
        self.label_2.setObjectName("label_2")
        self.k_label = QtWidgets.QLabel(self.Edge_Weld)
        self.k_label.setGeometry(QtCore.QRect(0, 430, 21, 18))
        self.k_label.setObjectName("k_label")
        self.g_label = QtWidgets.QLabel(self.Edge_Weld)
        self.g_label.setGeometry(QtCore.QRect(0, 350, 21, 18))
        self.g_label.setObjectName("g_label")
        self.g = QtWidgets.QTextEdit(self.Edge_Weld)
        self.g.setGeometry(QtCore.QRect(30, 340, 51, 31))
        self.g.setObjectName("g")
        self.e = QtWidgets.QTextEdit(self.Edge_Weld)
        self.e.setGeometry(QtCore.QRect(30, 300, 51, 31))
        self.e.setObjectName("e")
        self.i = QtWidgets.QTextEdit(self.Edge_Weld)
        self.i.setGeometry(QtCore.QRect(30, 380, 51, 31))
        self.i.setObjectName("i")
        self.e_label = QtWidgets.QLabel(self.Edge_Weld)
        self.e_label.setGeometry(QtCore.QRect(0, 310, 21, 18))
        self.e_label.setObjectName("e_label")
        self.i_label = QtWidgets.QLabel(self.Edge_Weld)
        self.i_label.setGeometry(QtCore.QRect(0, 390, 21, 18))
        self.i_label.setObjectName("i_label")
        self.k = QtWidgets.QTextEdit(self.Edge_Weld)
        self.k.setGeometry(QtCore.QRect(30, 420, 51, 31))
        self.k.setObjectName("k")
        self.label_3 = QtWidgets.QLabel(self.Edge_Weld)
        self.label_3.setGeometry(QtCore.QRect(20, 270, 161, 20))
        self.label_3.setObjectName("label_3")
        self.j = QtWidgets.QTextEdit(self.Edge_Weld)
        self.j.setGeometry(QtCore.QRect(150, 380, 51, 31))
        self.j.setObjectName("j")
        self.h = QtWidgets.QTextEdit(self.Edge_Weld)
        self.h.setGeometry(QtCore.QRect(150, 340, 51, 31))
        self.h.setObjectName("h")
        self.h_label = QtWidgets.QLabel(self.Edge_Weld)
        self.h_label.setGeometry(QtCore.QRect(120, 350, 21, 18))
        self.h_label.setObjectName("h_label")
        self.j_label = QtWidgets.QLabel(self.Edge_Weld)
        self.j_label.setGeometry(QtCore.QRect(120, 390, 21, 18))
        self.j_label.setObjectName("j_label")
        self.f_label = QtWidgets.QLabel(self.Edge_Weld)
        self.f_label.setGeometry(QtCore.QRect(120, 310, 21, 18))
        self.f_label.setObjectName("f_label")
        self.f = QtWidgets.QTextEdit(self.Edge_Weld)
        self.f.setGeometry(QtCore.QRect(150, 300, 51, 31))
        self.f.setObjectName("f")
        self.label_4 = QtWidgets.QLabel(self.Edge_Weld)
        self.label_4.setGeometry(QtCore.QRect(10, 490, 191, 20))
        self.label_4.setObjectName("label_4")
        self.m = QtWidgets.QTextEdit(self.Edge_Weld)
        self.m.setGeometry(QtCore.QRect(150, 520, 51, 31))
        self.m.setObjectName("m")
        self.m_label = QtWidgets.QLabel(self.Edge_Weld)
        self.m_label.setGeometry(QtCore.QRect(120, 530, 21, 18))
        self.m_label.setObjectName("m_label")
        self.l_label = QtWidgets.QLabel(self.Edge_Weld)
        self.l_label.setGeometry(QtCore.QRect(0, 530, 21, 18))
        self.l_label.setObjectName("l_label")
        self.l = QtWidgets.QTextEdit(self.Edge_Weld)
        self.l.setGeometry(QtCore.QRect(30, 520, 51, 31))
        self.l.setObjectName("l")
        self.stackedWidget.addWidget(self.Edge_Weld)
        self.Tekken = QtWidgets.QWidget()
        self.Tekken.setObjectName("Tekken")
        self.label_7 = QtWidgets.QLabel(self.Tekken)
        self.label_7.setGeometry(QtCore.QRect(20, 50, 321, 141))
        self.label_7.setObjectName("label_7")
        self.stackedWidget.addWidget(self.Tekken)
        self.button_ok = QtWidgets.QPushButton(Form)
        self.button_ok.setGeometry(QtCore.QRect(720, 620, 95, 26))
        self.button_ok.setObjectName("button_ok")
        self.button_save = QtWidgets.QPushButton(Form)
        self.button_save.setGeometry(QtCore.QRect(600, 620, 95, 26))
        self.button_save.setObjectName("button_save")

        self.retranslateUi(Form)
        self.stackedWidget.setCurrentIndex(2)
        QtCore.QMetaObject.connectSlotsByName(Form)

    def retranslateUi(self, Form):
        _translate = QtCore.QCoreApplication.translate
        Form.setWindowTitle(_translate("Form", "Form"))
        self.weld_label.setText(_translate("Form", "Weld Geometry:"))
        self.weld.setItemText(0, _translate("Form", "Edge Welded Beam"))
        self.weld.setItemText(1, _translate("Form", "TG9"))
        self.weld.setItemText(2, _translate("Form", "Tekken"))
        self.label.setText(_translate("Form", "Please select a Weld Geometry...."))
        self.b_label_2.setText(_translate("Form", "Weld Pass Length (B)"))
        self.a_label_2.setText(_translate("Form", "Weld Pass Thickness (A)"))
        self.c_label_2.setText(_translate("Form", "Weld Pass Height (C)"))
        self.label_5.setText(_translate("Form", "Geometry (mm):"))
        self.d_label_2.setText(_translate("Form", "Base plate Thickness (D)"))
        self.f_label_2.setText(_translate("Form", "Base plate Height (F)"))
        self.e_label_2.setText(_translate("Form", "Base plate Length (E)"))
        self.label_6.setText(_translate("Form", "Meshing (mm):"))
        self.c_label.setText(_translate("Form", "Weld Pass Height (C)"))
        self.label_1.setText(_translate("Form", "Geometry (mm):"))
        self.a_label.setText(_translate("Form", "Weld Pass Thickness (A)"))
        self.b_label.setText(_translate("Form", "Weld Pass Length (B)"))
        self.d_label.setText(_translate("Form", "Base plate Height (D)"))
        self.label_2.setText(_translate("Form", "Meshing:"))
        self.k_label.setText(_translate("Form", "K"))
        self.g_label.setText(_translate("Form", "G"))
        self.e_label.setText(_translate("Form", "E"))
        self.i_label.setText(_translate("Form", "I"))
        self.label_3.setText(_translate("Form", "Element length (mm):"))
        self.h_label.setText(_translate("Form", "H"))
        self.j_label.setText(_translate("Form", "J"))
        self.f_label.setText(_translate("Form", "F"))
        self.label_4.setText(_translate("Form", "Refined Zones length (mm):"))
        self.m_label.setText(_translate("Form", "K"))
        self.l_label.setText(_translate("Form", "J"))
        self.label_7.setText(_translate("Form", "Add tekken option..."))
        self.button_ok.setText(_translate("Form", "Ok"))
        self.button_save.setText(_translate("Form", "Save"))


if __name__ == "__main__":
    import sys
    app = QtWidgets.QApplication(sys.argv)
    Form = QtWidgets.QWidget()
    ui = Ui_ModuleMeshing()
    ui.setupUi(Form)
    Form.show()
    sys.exit(app.exec_())

